#!/usr/bin/env python
PACKAGE = "my_msgs"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("int_param", int_t, 0, "An Integer parameter", 50, 0, 100)
gen.add("double_param", double_t, 0, "A Double parameter", .5, 0, 1)
gen.add("str_param", str_t, 0, "A String parameter", "Hello World")
gen.add("bool_param", bool_t, 0, "A Boolean parameter", True)

#camera params
gen.add("is_large_resolution", bool_t, 0, "is MV CAM in large resolution mode", True)
gen.add("is_auto_exp", bool_t, 0, "is MV CAM in auto exposure mode", False)
gen.add("is_record", bool_t, 0, "is MV CAM recording video", False)
gen.add("exp_time", int_t, 0, "exposure time", 3000, 1000, 20000)

#threthold params
is_tgt_red=False
red_hmin_default=14
red_hmax_default=240
blue_hmin_default=220
blue_hmax_default=255
smin_default=40
smax_default=255
vmin_default=200
vmax_default=255
gen.add("is_red", bool_t,0,"the enemy is in red", is_tgt_red)
gen.add("v_min",int_t,0,"vis min",vmin_default)
gen.add("v_max",int_t,0,"vis max",vmax_default)
gen.add("s_min",int_t,0,"saturate min",smin_default)
gen.add("s_max",int_t,0,"saturate max",smax_default)
if(is_tgt_red):
    gen.add("h_min", int_t,0,"hue min",red_hmin_default)
    gen.add("h_max", int_t,0,"hue max",red_hmax_default)
else:
    gen.add("h_min", int_t,0,"hue min",blue_hmin_default)
    gen.add("h_max", int_t,0,"hue max",blue_hmax_default)

size_enum = gen.enum([gen.const("Small", int_t, 0, "A small constant"),
                      gen.const("Medium", int_t, 1, "A medium constant"),
                      gen.const("Large", int_t, 2, "A large constant"),
                      gen.const("Extralarge", int_t, 3, "A extra large constant")],
                      "An enum to set size")
gen.add("size", int_t, 0, "A size parameter which is edited via an enum", 1, 0, 3, edit_method=size_enum)

exit(gen.generate(PACKAGE, "my_msgs", "tutorials"))

